### PR TITLE
Pa/oauth testing2

### PR DESCRIPTION
**Pull request by palworth**
## Description
Completes User Story #
Resolves #

## Type of change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Refactor (non-breaking change which maintains existing functionality)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Notes
Added spec for Oauth login, utilized oauth mock to replicate hash returned from oauth
## RSpec Results
```

Finished in 0.39306 seconds (files took 2.82 seconds to load)
6 examples, 0 failures

```

### CODE REVIEW

File: app/controllers/sessions_controller.rb [hunk 1]
```diff
@@ -1,13 +1,6 @@
...
-  # def create 
-
-  #   render plain: request.env["omniauth.auth"]
-  #   # request.env['omniauth.auth']
-  #   # render text: request.env['omniauth.auth'].inspect
-  # end
-
   def create
-    if user = User.from_omniauth(request.env["omniauth.auth"])
+    if user = User.update_or_create(request.env["omniauth.auth"])
```
Comment (palworth) [inline app/controllers/sessions_controller.rb:3]: change form_omniauth method to update_or_create for readability

File: app/models/user.rb [hunk 2]
```diff
@@ -3,12 +3,25 @@ class User < ApplicationRecord
...
+  #   binding.pry
+  #   where(uid: auth_info[:uid]).first_or_create do |new_user|
+  #     new_user.uid                 = auth_info.uid
+  #     new_user.name                = auth_info.extra.raw_info.name
+  #     new_user.email               = auth_info.extra.raw_info.email
+  #     new_user.google_token        = auth_info.credentials.token
+  #   end
+  # end 
+
+  def self.update_or_create(auth_info)
```
Comment (palworth) [inline app/models/user.rb:16]: Refactored method for readability.

File: spec/features/user_logs_in_with_google_spec.rb [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+        click_link 'Login with Google'
+
+        expect(page).to have_content("Jack Black")
+        expect(page).to have_link("Logout")
+    end 
+end 
+
+def stub_omniauth
+  OmniAuth.config.test_mode = true
+  OmniAuth.config.mock_auth[:google_oauth2] = OmniAuth::AuthHash.new({
```
Comment (palworth) [inline spec/features/user_logs_in_with_google_spec.rb:17]: OmniAuth.config.mock_auth[:google_oauth2] configures the hash specific for google oauth2 gem.

File: app/models/user.rb (hunk 2 above)
Comment (cgaddis36) [reply app/models/user.rb:16]: nice, I like this a lot more

File: spec/features/user_logs_in_with_google_spec.rb (hunk 3 above)
Comment (cgaddis36) [reply spec/features/user_logs_in_with_google_spec.rb:17]: very cool, never seen a method below a test before; but I like it

**Pull request merged**